src/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1486,5 +1478,118 @@ private bool TryGetHostUri(string hostName, out Uri hostUri)
             string s = Address.Scheme + "://" + hostName + Address.PathAndQuery;
             return Uri.TryCreate(s, UriKind.Absolute, out hostUri);
         }
+
+        private HttpClient GetCachedOrCreateHttpClient(out bool disposeRequired)
+        {
+            if (!HttpClientParameters.AreParametersAcceptableForCaching(this))
+            {
+                disposeRequired = true;
+                return CreateHttpClient();
+            }
+
+            disposeRequired = false;
+            if (s_cachedHttpClient == null)
+            {
+                lock (s_syncRoot)
+                {
+                    if (s_cachedHttpClient == null)
+                    {
+                        s_cachedHttpClientParameters = new HttpClientParameters(this);
+                        s_cachedHttpClient = CreateHttpClient();
+                        return s_cachedHttpClient;
+                    }
+                }
+            }
+
+            if (s_cachedHttpClientParameters.Matches(this))
+            {
+                return s_cachedHttpClient;
+            }
+
+            disposeRequired = true;
+            return CreateHttpClient();
```

Comment:
These two lines are identical to the two that start the method.  It can instead be structured as:
```C#
if (HttpClientParameters.AreParametersAcceptableForCaching(this))
{
    ... // everything in the middle of this method
}

disposeRequired = true;
return CreateHttpClient();
```

Reply:
Fixed